karapace/master_coordinator.py

```diff
@@ -236,7 +238,11 @@ def run(self) -> None:
                 self.sc.ensure_active_group()
                 self.sc.poll_heartbeat()
                 LOG.debug("We're master: %r: master_uri: %r", self.sc.are_we_master, self.sc.master_url)
-                time.sleep(min(_hb_interval, self.sc.time_to_next_heartbeat()))
+                # In cases when heartbeat is missed the sleep min sleep time would be 0
+                # from `time_to_next_heartbeat`. In that case halve the heartbeat interval for
+                # some sane sleep instead of running the loop without sleep for a while.
+                sleep_time = min(_hb_interval, self.sc.time_to_next_heartbeat()) or _hb_interval / 2
```

Comment:
Could this be written as an `if`-statement for readability reasons?

Reply:
Done.